modules/mono/mono_gd/gd_mono_marshal.cpp

```diff
@@ -1112,8 +1111,7 @@ Variant mono_object_to_variant_impl(MonoObject *p_obj, const ManagedType &p_type
 			if (CACHED_CLASS(GodotObject)->is_assignable_from(type_class)) {
 				Object *ptr = unbox<Object *>(CACHED_FIELD(GodotObject, ptr)->get_value(p_obj));
 				if (ptr != NULL) {
-					Reference *ref = Object::cast_to<Reference>(ptr);
-					return ref ? Variant(Ref<Reference>(ref)) : Variant(ptr);
+					return ptr->is_reference() ? Variant(Ref<Reference>(ptr)) : Variant(ptr);
```

Comment:
I'm a bit surprised you don't have to cast in some of these cases. Is the ptr already `Reference` compatible? Or does it somehow auto cast?

Reply:
I was more concerned that it's statically possible at all. I would have expected it to require an explicit pointer static cast.